src/Core/src/Platform/Android/ViewExtensions.cs

```diff
@@ -185,7 +181,24 @@ public static void SetClipToOutline(this AView view, bool value)
 
 		public static void UpdateAutomationId(this AView nativeView, IView view)
 		{
-			nativeView.SetTag(AutomationTagId, view.AutomationId);
+			if (!string.IsNullOrWhiteSpace(view.AutomationId))
+			{
+				nativeView = nativeView.GetSemanticNativeElement();
+				var importantForAccessibility = nativeView.ImportantForAccessibility;
+				nativeView.ContentDescription = view.AutomationId;
+
+				// Android automatically sets ImportantForAccessibility to "Yes" when you set the ContentDescription.
+				// We are only setting the ContentDescription here for Automation testing purposes so
+				// we don't want Layouts/images/Other controls to automatically toggle to Yes.
+				// Unfortunately Android (AFAICT) doesn't have an obvious way of calculating what "Auto" will be interpreted as
+				// iOS is kind of enough to indicate that anything inheriting from "UIControl" but the Android documentation
+				// just says "Android uses heuristics to figure out what Auto will mean"
+				// It seems like if we just toggle this back to "Auto" that everything just works.
+				if (importantForAccessibility == ImportantForAccessibility.Auto)
+				{
+					nativeView.ImportantForAccessibility = ImportantForAccessibility.Auto;
+				}
```

Comment:
Will this be called a lot? Could this be something where you have a `Java` method like:

```java
public static void setContentDescription(View view, String description)
{
    int importantForAccessibility = view.getImportantForAccessibility();
    view.setContentDescription(description);
    if (importantForAccessibility == View.IMPORTANT_FOR_ACCESSIBILITY_AUTO) {
        view.setImportantForAccessibility(View.IMPORTANT_FOR_ACCESSIBILITY_AUTO);
    }
}
```
This takes 3 JNI calls down to 1, you'd use it like:

```csharp
ViewHelper.SetContentDescription(nativeView.GetSemanticNativeElement(), view.AutomationId);
```

Unless we can also move the `GetSemanticNativeElement()` logic into Java.